In [2]:
import pandas as pd
import numpy as np

print("pandas version :", pd.__version__)
print("numpy version  :", np.__version__)
print("Library siap digunakan!")

pandas version : 2.2.2
numpy version  : 2.0.2
Library siap digunakan!


In [3]:
# ── LIST ──
data = [10, 20, 'Python', True, 3.14]
print("List:", data)
print("Akses index 0:", data[0])
print("Slicing [1:3]:", data[1:3])

# ── TUPLE ──
koordinat = (6.9175, 107.6191)  # Bandung
print("\nTuple koordinat:", koordinat)

# ── DICTIONARY ──
mahasiswa = {
    'nama': 'Budi Santoso',
    'nim': '2024001',
    'prodi': 'Informatika',
    'ipk': 3.85
}
print("\nDictionary mahasiswa:", mahasiswa)
print("Nama:", mahasiswa['nama'])

# ── SET ──
nilai = {85, 90, 85, 78, 90, 95}
print("\nSet (duplikat dihapus otomatis):", nilai)

List: [10, 20, 'Python', True, 3.14]
Akses index 0: 10
Slicing [1:3]: [20, 'Python']

Tuple koordinat: (6.9175, 107.6191)

Dictionary mahasiswa: {'nama': 'Budi Santoso', 'nim': '2024001', 'prodi': 'Informatika', 'ipk': 3.85}
Nama: Budi Santoso

Set (duplikat dihapus otomatis): {90, 85, 78, 95}


In [4]:
# ── MEMBUAT ARRAY ──
a1D = np.array([1, 2, 3, 4, 5])
a2D = np.array([[1,2,3],[4,5,6]])

print("Array 1D:", a1D)
print("Array 2D:\n", a2D)
print("Shape:", a2D.shape)
print("Dtype:", a2D.dtype)

# ── OPERASI VEKTORISASI ──
a = np.array([1, 2, 3, 4, 5])
b = np.array([10, 20, 30, 40, 50])
print("\na + b =", a + b)
print("a * b =", a * b)
print("a ** 2 =", a ** 2)

# ── STATISTIK ──
rng = np.random.default_rng(seed=42)
data = rng.normal(0, 1, 1000)
print(f"\nMean : {np.mean(data):.4f}")
print(f"Std  : {np.std(data):.4f}")
print(f"Min  : {np.min(data):.4f}")
print(f"Max  : {np.max(data):.4f}")

Array 1D: [1 2 3 4 5]
Array 2D:
 [[1 2 3]
 [4 5 6]]
Shape: (2, 3)
Dtype: int64

a + b = [11 22 33 44 55]
a * b = [ 10  40  90 160 250]
a ** 2 = [ 1  4  9 16 25]

Mean : -0.0289
Std  : 0.9887
Min  : -3.6484
Max  : 3.1789


In [5]:
# ── PANDAS SERIES ──
nilai = pd.Series(
    [85, 92, 78, 95],
    index=['Alice', 'Bob', 'Caca', 'Dina']
)
print("Series nilai:")
print(nilai)
print("\nNilai Bob:", nilai['Bob'])
print("Filter > 80:\n", nilai[nilai > 80])
print("Mean:", nilai.mean())
print("\nDescribe:")
print(nilai.describe())

# ── PANDAS DATAFRAME ──
data = {
    'nama': ['Ali', 'Budi', 'Caca', 'Dina'],
    'usia': [22, 25, 21, 23],
    'kota': ['Jakarta', 'Bandung', 'Surabaya', 'Medan'],
    'ipk': [3.8, 3.5, 3.9, 3.7]
}
df = pd.DataFrame(data)
print("\nDataFrame mahasiswa:")
print(df)
print("\nShape:", df.shape)
print("Dtypes:\n", df.dtypes)

Series nilai:
Alice    85
Bob      92
Caca     78
Dina     95
dtype: int64

Nilai Bob: 92
Filter > 80:
 Alice    85
Bob      92
Dina     95
dtype: int64
Mean: 87.5

Describe:
count     4.000000
mean     87.500000
std       7.593857
min      78.000000
25%      83.250000
50%      88.500000
75%      92.750000
max      95.000000
dtype: float64

DataFrame mahasiswa:
   nama  usia      kota  ipk
0   Ali    22   Jakarta  3.8
1  Budi    25   Bandung  3.5
2  Caca    21  Surabaya  3.9
3  Dina    23     Medan  3.7

Shape: (4, 4)
Dtypes:
 nama     object
usia      int64
kota     object
ipk     float64
dtype: object


In [6]:
# ── LOAD DATASET TITANIC ──
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

# Eksplorasi standar
print("=== HEAD (5 baris pertama) ===")
display(df.head())

print("\n=== INFO ===")
df.info()

print("\n=== DESCRIBE ===")
display(df.describe())

=== HEAD (5 baris pertama) ===


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S



=== INFO ===
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB

=== DESCRIBE ===


,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [7]:
# ── Q1: Struktur Dataset Titanic ──

print("Jumlah baris  :", df.shape[0])
print("Jumlah kolom  :", df.shape[1])
print("\nNama kolom:")
print(df.columns.tolist())

Jumlah baris  : 891
Jumlah kolom  : 12

Nama kolom:
['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


## Q1 — Struktur Dataset Titanic
Dataset Titanic memiliki **891 baris** dan **12 kolom**.
Kolom yang tersedia: PassengerId, Survived, Pclass, Name, Sex, Age, SibSp, Parch, Ticket, Fare, Cabin, Embarked.

In [8]:
# ── Q2: Missing Values ──

missing_count = df.isnull().sum()
missing_pct = (df.isnull().sum() / len(df) * 100).round(2)

missing_df = pd.DataFrame({
    'Jumlah Missing': missing_count,
    'Proporsi (%)': missing_pct
})

# Tampilkan hanya kolom yang ada missing values
print(missing_df[missing_df['Jumlah Missing'] > 0])

          Jumlah Missing  Proporsi (%)
Age                  177         19.87
Cabin                687         77.10
Embarked               2          0.22


## Q2 — Missing Values
Terdapat 3 kolom dengan missing values:
- **Cabin**: 687 missing (77.10%) — sangat banyak
- **Age**: 177 missing (19.87%) — perlu diperhatikan
- **Embarked**: 2 missing (0.22%) — dapat diabaikan

In [9]:
# ── Q3: Persentase Penumpang Selamat ──

total = len(df)
selamat = df['Survived'].sum()
pct_selamat = (selamat / total * 100).round(2)

print("Total penumpang :", total)
print("Jumlah selamat  :", selamat)
print("Tidak selamat   :", total - selamat)
print(f"Persentase selamat: {pct_selamat}%")

Total penumpang : 891
Jumlah selamat  : 342
Tidak selamat   : 549
Persentase selamat: 38.38%


## Q3 — Persentase Penumpang Selamat
Hanya **38.38%** (342 dari 891) penumpang yang selamat.
Sebanyak 549 penumpang (61.62%) tidak selamat.
Hal ini sesuai fakta sejarah karena jumlah sekoci Titanic
tidak mencukupi untuk semua penumpang.

In [10]:
# ── Q4: Penumpang Wanita Kelas 1 ──

wanita_kelas1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]

print("Jumlah penumpang wanita kelas 1:", len(wanita_kelas1))
print("Rata-rata usia                 :", wanita_kelas1['Age'].mean().round(2))

Jumlah penumpang wanita kelas 1: 94
Rata-rata usia                 : 34.61


## Q4 — Penumpang Wanita Kelas 1
Terdapat **94 penumpang wanita** di kelas 1 dengan rata-rata usia **34.61 tahun**.
Wanita kelas 1 merupakan kalangan atas yang kemungkinan besar
mendapat prioritas evakuasi ("women and children first").

In [11]:
# ── Q5: Tingkat Keselamatan per Kelas ──

survival_per_kelas = df.groupby('Pclass')['Survived'].mean().round(4) * 100

print("Tingkat keselamatan per kelas (%):")
print(survival_per_kelas)
print("\nKelas dengan tingkat keselamatan tertinggi: Kelas", survival_per_kelas.idxmax())

Tingkat keselamatan per kelas (%):
Pclass
1    62.96
2    47.28
3    24.24
Name: Survived, dtype: float64

Kelas dengan tingkat keselamatan tertinggi: Kelas 1


## Q5 — Tingkat Keselamatan per Kelas
Kelas 1: 62.96% | Kelas 2: 47.28% | Kelas 3: 24.24%
**Kelas 1 memiliki tingkat keselamatan tertinggi.**
Penumpang kelas atas mendapat akses lebih mudah ke sekoci
dibanding penumpang kelas bawah.

In [12]:
# ── Q6: Tingkat Keselamatan Anak-anak vs Dewasa ──

# Hapus data yang Age-nya kosong
df_age = df.dropna(subset=['Age'])

# Kategorikan anak-anak dan dewasa
df_age = df_age.copy()
df_age['Kategori'] = df_age['Age'].apply(lambda x: 'Anak-anak (<18)' if x < 18 else 'Dewasa (>=18)')

# Hitung tingkat keselamatan
hasil = df_age.groupby('Kategori')['Survived'].agg(
    Jumlah='count',
    Selamat='sum',
).reset_index()
hasil['Pct_Selamat (%)'] = (hasil['Selamat'] / hasil['Jumlah'] * 100).round(2)

print("Tingkat Keselamatan Anak-anak vs Dewasa:")
print(hasil.to_string(index=False))

Tingkat Keselamatan Anak-anak vs Dewasa:
       Kategori  Jumlah  Selamat  Pct_Selamat (%)
Anak-anak (<18)     113       61            53.98
  Dewasa (>=18)     601      229            38.10


## Q6 — Anak-anak vs Dewasa
Anak-anak (<18 tahun): **53.98%** selamat
Dewasa (>=18 tahun): **38.10%** selamat

Anak-anak memiliki tingkat keselamatan 15% lebih tinggi dari dewasa.
Hal ini membuktikan kebijakan evakuasi "women and children first"
benar-benar diterapkan saat tragedi Titanic tenggelam.